src/lib/eliom_client.client.ml

```diff
@@ -1065,8 +1066,9 @@ let change_page (type m)
                (flush_onchangepage ())
            in
            with_new_page ~replace () @@ fun () ->
+           let%lwt () = f get_params post_params in
```

Comment:
I'm quite sure that swapping these two lines breaks client-side redirection. Indeed, they are implemented by calling `change_page` from function `f`, replacing the current page. If you change the url after the call to `f`, the wrong page will be replaced.